pkg/cmd/generator/certs/main.go

```diff
@@ -0,0 +1,113 @@
+package certs
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+)
+
+//go:embed tls
+var ownershipFiles embed.FS
+
+type PKIOwnerInfo struct {
+	Owners []string `json:"owners"`
+	// +mapType:=atomic
+	CertificateAuthorityBundles []certgraphapi.PKIRegistryInClusterCABundle `json:"certificateAuthorityBundles"`
+	// +mapType:=atomic
+	CertKeyPairs []certgraphapi.PKIRegistryInClusterCertKeyPair `json:"certKeyPairs"`
+}
+
+func CombineAllOwnershipFiles() ([]PKIOwnerInfo, error) {
+	result := []PKIOwnerInfo{}
+	files, err := listChunkFiles()
+	if err != nil {
+		return result, fmt.Errorf("failed to list owner files: %v", err)
+	}
+	for _, fileName := range files {
+		contents, err := ownershipFiles.ReadFile(filepath.Join("tls", fileName))
+		if err != nil {
+			return result, fmt.Errorf("failed to read file %s: %v", fileName, err)
+		}
+		chunkOwnerInfo := PKIOwnerInfo{}
+		err = json.Unmarshal(contents, &chunkOwnerInfo)
+		if err != nil {
+			return result, fmt.Errorf("error unmarshalling file %s: %v", fileName, err)
+		}
+		result = append(result, chunkOwnerInfo)
+	}
+	return result, nil
+}
+
+func validateChunk(ownerInfo *PKIOwnerInfo) error {
+	if len(ownerInfo.Owners) == 0 {
+		return fmt.Errorf("no owners specified")
+	}
+
+	for _, certKeyPair := range ownerInfo.CertKeyPairs {
+		if certKeyPair.SecretLocation.Name == "" {
+			return fmt.Errorf("name not set: %#v", certKeyPair)
+		}
+		if certKeyPair.SecretLocation.Namespace == "" {
+			return fmt.Errorf("namespace not set: %#v", certKeyPair)
+		}
+		if certKeyPair.CertKeyInfo.OwningJiraComponent == "" {
+			return fmt.Errorf("jira component not set: %#v", certKeyPair)
+		}
+		if certKeyPair.CertKeyInfo.Description == "" {
+			return fmt.Errorf("description not set: %#v", certKeyPair)
+		}
+	}
+
+	for _, caBundle := range ownerInfo.CertificateAuthorityBundles {
+		if caBundle.ConfigMapLocation.Name == "" {
+			return fmt.Errorf("name not set: %#v", caBundle)
+		}
+		if caBundle.ConfigMapLocation.Namespace == "" {
+			return fmt.Errorf("namespace not set: %#v", caBundle)
+		}
+		if caBundle.CABundleInfo.OwningJiraComponent == "" {
+			return fmt.Errorf("jira component not set: %#v", caBundle)
+		}
+		if caBundle.CABundleInfo.Description == "" {
+			return fmt.Errorf("description not set: %#v", caBundle)
+		}
+	}
+
+	return nil
+}
+
+func ValidateTLSOwnership(outputDir string) error {
+	var errors []error
+
+	owners, err := CombineAllOwnershipFiles()
+	if err != nil {
+		return err
+	}
+	for _, chunk := range owners {
+		if err := validateChunk(&chunk); err != nil {
```

Comment:
when this gets used in an e2e test, we want to be sure that existing violators (our current state) don't fail the test.  I think that requires
1. know which secrets and configmaps lack owners
2. know which secrets and confimaps in the cluster lack owners
3. subtract the known from current, and any new ones will fail.

Reply:
1. `verify-tls-owners` will ensure that all registred artifacts have owners set
2 and 3. would show up as unregistered artifact:
```
 Unregistered TLS artifact: --namespace=openshift-etcd, secret/etcd-serving-ip-10-0-94-141.us-east-2.compute.internal
```